kinto/core/cache/postgresql/__init__.py

```diff
@@ -8,6 +9,17 @@
 from kinto.core.utils import json
 
 
+def retry_on_failure(func):
+    @wraps(func)
+    def wraps_func(self, *args, **kwargs):
+        import psycopg2
```

Comment:
I we put it on the top of the file it will fail in case the psycopg2 driver is not installed.